File: src/main/scala/com/test/scala/business/template/WordSplitBolt.scala
package com.test.scala.business.template

import backtype.storm.topology.{BasicOutputCollector, OutputFieldsDeclarer}
import backtype.storm.topology.base.BaseBasicBolt
import backtype.storm.tuple.{Fields, Tuple, Values}

class WordSplitBolt extends BaseBasicBolt{
  override def execute(input: Tuple, collector: BasicOutputCollector): Unit = {
    val lineStr = input.getString(0)
    val wordArr = lineStr.split(" ")

    for(word <- wordArr){
      collector.emit(new Values(word))
    }
  }

  override def declareOutputFields(declarer: OutputFieldsDeclarer): Unit = {
    declarer.declare(new Fields("WORD"))
  }
}

File: src/main/java/com/test/java/business/template/WordSplitBolt.java
package com.test.java.business.template;

import backtype.storm.topology.BasicOutputCollector;
import backtype.storm.topology.OutputFieldsDeclarer;
import backtype.storm.topology.base.BaseBasicBolt;
import backtype.storm.tuple.Fields;
import backtype.storm.tuple.Tuple;
import backtype.storm.tuple.Values;

/**
 * @author lvxw
 */
public class WordSplitBolt extends BaseBasicBolt {

    @Override
    public void execute(Tuple input, BasicOutputCollector collector) {
        String lineStr = input.getString(0);
        String[] wordArr = lineStr.split(" ");
        for(String word:wordArr){
            collector.emit(new Values(word));
        }

    }

    @Override
    public void declareOutputFields(OutputFieldsDeclarer declarer) {
        declarer.declare(new Fields("WORD"));
    }
}
